### ISSUE TITLE
Strange Behavior on Wayland + gnome

### ISSUE THREAD

**Issue by haydonryan** [federico-terzi/espanso#1893]
**Describe the bug**
Espanso causes keyboard to miss keystrokes and the active window to flash intermittently in wayland

**To Reproduce**
1. Install Manually following instructions for Manjaro / Arch on Arch with espanso.
2. Start typing in any program

**Expected behavior**
I shouldn't get missing keystrokes and the focus leaving the current window.

**Screenshots**
If applicable, add screenshots or screen recordings to help explain your problem.

**Logs**
If possible, run `espanso log` in a terminal after the bug has occurred, then post the output here so that we can better diagnose the problem

**Setup information**
- OS: 6.7.8-arch1-1 with gnome 45.4.  
- Version: 2.2.1

**Comment by LeeRoyManea** [comment]
Same problem here on Fedora 40. 
When i went to a Terminal and start typing everything works. 
But after switching to Jetbrains Rider, the cursor is still in the terminal. 
So i cant write in Rider.
I have to switch to Chrome / Firefox and switch than to Jerbrans Rider. 
But these are not the only applications i have trouble with

https://github.com/espanso/espanso/assets/114137402/b481849c-323b-4a4c-b443-3d75c50dc9a2

**Comment by smeech** [comment]
The flashing/extraneous window in Wayland is a well documented bug.

Have you tried the [`force_mode:`](https://espanso.org/docs/matches/basics/#injection-mechanism) options on your expansions?

**Comment by haydonryan** [comment]
No I haven't - will try that and do more testing.  Thanks for the heads up on the other bug.  I'll test more, and wat for the PR to get merged into a release.

**Comment by haydonryan** [comment]
Coming back to this almost two years later... 

Is wayland working well without the flashing window? I did end up trying force_mode: a long while ago, but had no luck so just uninstalled and dealt with having to type everything.

**Comment by smeech** [comment]
> Is wayland working well without the flashing window?

Yes, I think that was fixed a few versions ago, and shouldn't be a problem with the current release. We still haven't any progress with [app-specific configurations](https://espanso.org/docs/configuration/app-specific-configurations/) or keyboard detection, however.

**Comment by haydonryan** [comment]
Just did a re-install according to the commands on the install page.   (Completely uptodate arch + cosmic setup).

Can confirm unless there's steps I'm supposed to do the window with the color palate comes up, and regular keys are not always responsive.  

Ok it looks like a required package wl-clipboard was not installed - however only looking at journalctl -f reported this.  Actually got it to text expand but it's not reliable - open a new termianl expanso doesn't work, but then it does in the original terminal.  Go to gmail works.  I def need to experiment more with this. .

Message to the devs - please have it check for required packages before it says install successful.
